spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/config/ConformanceConfigInstance.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.config
+
+import org.apache.spark.storage.StorageLevel
+import scopt.OParser
+import za.co.absa.enceladus.common.config.{ConfigError, JobConfig}
+import za.co.absa.enceladus.dao.auth.{InvalidMenasCredentialsFactory, MenasCredentialsFactory}
+
+import scala.util.Try
+
+
+/**
+ * This is a class for configuration provided by the command line parameters
+ *
+ * Note: scopt requires all fields to have default values.
+ *       Even if a field is mandatory it needs a default value.
+ */
+case class ConformanceConfigInstance(datasetName: String = "",
```

Comment:
I have been thinking about it and calling a non-singleton class a `SomethingInstance` seems a tad weird to me. Wouldn't `ConformanceConfig` suffice?

Reply:
You have a good point regarding the `Instance` suffix.
`Parser` suffix looks good. 👍 
My problem with conjunctive usage of `ConformanceConfig` and `ConformanceCmdConfig`, sytacically they are very easy to mix up (in understanding) and doesn't says anything about their relation.

Reply:
Let me go on a limb again and say, do we need the full name? It is in package `...conformance.config` shouldn't `...conformance.config.Parser` suffice? We are not gonna use it out of the scope of Conformance anyway.

Reply:
Having a PropertiesProvider for each package makes sense, but in the case of the config not so much because both of them will be used further on

Reply:
You are right. Let's see when the jobs merge